allow_cidr/middleware.py

```diff
@@ -47,6 +48,9 @@ def __call__(self, request):
 
         if not domain or not validate_host(domain, ORIG_ALLOWED_HOSTS):
             should_raise = True
+            if domain[-1] == "]":
```

Comment:
Probably unlikely in reality, but still an edge case: this might end up as `None[-1]` which would be an unhappy moment. 
```suggestion
            if domain and domain[-1] == "]":
```

And (again  in the super-edge-case zone) would be nice to check first and last, given we're about to drop them from the string. Your call whether you think this feels right

```suggestion
            if domain and domain[0] == "[" and domain[-1] == "]":
```

Reply:
Agreed, domain should be checked for None or empty.

I went with how Django checks if the host is an IPv6: https://github.com/django/django/blob/main/django/http/request.py#L706

Reply:
Cool - have merged the edit and run the tests. Just waiting for an extra perspective from @robhudson